### PR TITLE
Remove deprecated percolate and mpercolate apis

### PR DESCRIPTION
**Pull request by martijnvg**

### CODE REVIEW

File: modules/percolator/src/test/java/org/elasticsearch/percolator/PercolatorQuerySearchIT.java [hunk 1]
```diff
@@ -605,4 +603,82 @@ public void testPercolateQueryWithNestedDocuments() throws Exception {
         assertHitCount(response, 0);
     }
 
+    public void testPercolatorQueryViaMultiSearch() throws Exception {
```
Comment (dadoonet) [inline modules/percolator/src/test/java/org/elasticsearch/percolator/PercolatorQuerySearchIT.java]: Does this should also go in 5.x branches actually?

**Review by dadoonet — COMMENTED** [review]
Nice stats. I left a comment about the tests you migrated / added.

File: modules/percolator/src/test/java/org/elasticsearch/percolator/PercolatorQuerySearchIT.java (hunk 1 above)
Comment (martijnvg) [inline modules/percolator/src/test/java/org/elasticsearch/percolator/PercolatorQuerySearchIT.java]: Good point. I'll move these tests separately to the 5.x branch.

File: modules/percolator/src/test/resources/rest-api-spec/test/mpercolate/10_basic.yaml [hunk 2]
```diff
@@ -1,56 +0,0 @@
----
-"Basic multi-percolate":
```
Comment (nik9000) [inline modules/percolator/src/test/resources/rest-api-spec/test/mpercolate/10_basic.yaml]: Does msearch need a test for this?

File: modules/percolator/src/test/resources/rest-api-spec/test/percolate/15_new.yaml [hunk 3]
```diff
@@ -1,46 +0,0 @@
----
-"Basic percolation tests":
-
-  - do:
```
Comment (nik9000) [inline modules/percolator/src/test/resources/rest-api-spec/test/percolate/15_new.yaml]: What about these? I didn't see them in 5.x but maybe I didn't look super closely.

File: modules/percolator/src/test/resources/rest-api-spec/test/mpercolate/10_basic.yaml (hunk 2 above)
Comment (martijnvg) [inline modules/percolator/src/test/resources/rest-api-spec/test/mpercolate/10_basic.yaml]: Agreed, I added a basic test for both the search and msearch apis using a `percolate` query.

File: modules/percolator/src/test/resources/rest-api-spec/test/percolate/15_new.yaml (hunk 3 above)
Comment (martijnvg) [inline modules/percolator/src/test/resources/rest-api-spec/test/percolate/15_new.yaml]: Agreed, I added a basic test for both the search and msearch apis using a `percolate` query.

**Comment by martijnvg** [comment]
Thanks for checking this out @nik9000 @dadoonet! I've updated the PR.

**Review by nik9000 — APPROVED** [review]

**Pull request merged**